cmake/IRODSConfigVersion.cmake.in

```diff
@@ -0,0 +1,170 @@
+# CMake package version file for iRODS, for use with find_package().
+#
+# A couple of details about our version numbers:
+#
+# Prior to iRODS version 5.0, major releases incremented the second part of the version number,
+# and minor releases incremented the third part. For example, 4.3.0 was a major release, and 4.3.1
+# was a minor release of 4.3.
+# Starting with iRODS 5.0, major releases increment the first part of the version number, and
+# minor releases increment the second part. For example, 5.0 is a major release, and 5.1 is a
+# minor release of 5. Patch releases, should we decide to do them, increment the third part of the
+# version number. For example, 5.0.1 would be a patch release of 5.0.
+# 4.3 was the last major release before 5.0.
+#
+# Additionally, starting with the leadup to iRODS version 5.0, we have adopted .9x version numbers
+# for development builds. For example, 4.97 would be a development version between 4.3 and 5.0,
+# and 5.92 would be a development version between 5 and 6.0.
+# Starting with iRODS 5.0, this extends to minor releases as well. For example, 5.1.94 would be a
+# development version between 5.1 and 5.2.0.
+#
+# Thusly, our CMake package version file is designed as so:
+# - An iRODS installation should match as exactly compatible if the find_package()-calling project
+#   is searching for the exact same version.
+# - An iRODS installation should match as compatible if the find_package()-calling project is
+#   searching for a version range containing the version of the iRODS installation.
+# - An iRODS installation without a .9x version should match as potentially compatible if the
+#   find_package()-calling project is searching for an earlier version of the same major release
+#   or a .9x version of the previous major release.
+# - An iRODS installation with a .9x version in the second part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   major release or any earlier version of the same major release.
+# - An iRODS installation with a .9x version in the third part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   minor release or any earlier version of the same major release.
+
+set(IRODS_VERSION_1 "@IRODS_VERSION_MAJOR@")
+set(IRODS_VERSION_2 "@IRODS_VERSION_MINOR@")
+set(IRODS_VERSION_3 "@IRODS_VERSION_PATCH@")
```

Comment:
> Why _1, _2, _3 here instead of something perhaps a bit more descriptive? Is it because "major", "minor", and "patch" have different meanings in 4.x and 5.x and this is just the easiest way to refer to the parts?

Yes, exactly.